client/replays/action-creators.ts

```diff
@@ -15,6 +17,23 @@ import { makeServerUrl } from '../network/server-url'
 
 const ipcRenderer = new TypedIpcRenderer()
 
+class ReplayFile {
```

Comment:
This really doesn't feel like it should be a class (and the code that follows and uses it is not technically correct, since the JS code that uses this elsewhere is *not* passing instances of this class). What you want is a simple interface:

```ts
interface ReplayFile {
  path: string
  name: string
}
```

No real value from the `fromPath` function here either since we only do that in a single place, you can just inline that implementation there